From 36e7a0ae19034bd58e5e74942d5f8b19697f5fa6 Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Mon, 4 Jul 2022 15:22:58 +0800
Subject: [PATCH] Revert "xfs: remove all COW fork extents when remounting
 readonly"
Patch-mainline: Never, revert stable patch which is added again in sorted section
References: bsn#197
Modified-by-SEL: No


This reverts commit 071e750ffb3dc625cc92826950c26554f161a32c.

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/xfs/xfs_super.c | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/fs/xfs/xfs_super.c b/fs/xfs/xfs_super.c
index 05cea7788d49..5ebd6cdc44a7 100644
--- a/fs/xfs/xfs_super.c
+++ b/fs/xfs/xfs_super.c
@@ -1695,10 +1695,7 @@ static int
 xfs_remount_ro(
 	struct xfs_mount	*mp)
 {
-	struct xfs_eofblocks	eofb = {
-		.eof_flags	= XFS_EOF_FLAGS_SYNC,
-	};
-	int			error;
+	int error;
 
 	/*
 	 * Cancel background eofb scanning so it cannot race with the final
@@ -1706,13 +1703,8 @@ xfs_remount_ro(
 	 */
 	xfs_stop_block_reaping(mp);
 
-	/*
-	 * Clear out all remaining COW staging extents and speculative post-EOF
-	 * preallocations so that we don't leave inodes requiring inactivation
-	 * cleanups during reclaim on a read-only mount.  We must process every
-	 * cached inode, so this requires a synchronous cache scan.
-	 */
-	error = xfs_icache_free_cowblocks(mp, &eofb);
+	/* Get rid of any leftover CoW reservations... */
+	error = xfs_icache_free_cowblocks(mp, NULL);
 	if (error) {
 		xfs_force_shutdown(mp, SHUTDOWN_CORRUPT_INCORE);
 		return error;
-- 
2.34.1

